### PR TITLE
Reuse node names when installing new k8s pods

### PR DESCRIPTION
**Pull request by spilchen**
This corrects node name reuse when the operator installs new hosts in the admintools.conf. The old reuse logic was a bit too simple and could end up skip reusing a name under certain circumstances. 

Note, there are two kinds of node names in vertica. The node name before a node is added to the cluster -- in the format of node####. And the vnode name, which is the name after its added to the cluster -- in the format of v_<dbname>_node####. This impacts only the first format.

### CODE REVIEW

File: pkg/atconf/file_writer.go [hunk 1]
```diff
@@ -293,13 +293,28 @@ func (f *FileWriter) getNextNodeNumber() int {
...
 	}
-	return nextNodeNumber
+	return nodes
+}
+
+// getNextNodeNumber returns the number to use for the next vertica node.  It
+// determines this by finding a free spot in the nodes slice.
+func (f *FileWriter) getNextNodeNumber(nodes []bool) (nextNodeNumber int, newNodes []bool) {
+	for i := 0; i < len(nodes); i++ {
+		if !nodes[i] {
```
Comment (roypaulin) [inline pkg/atconf/file_writer.go:310]: I don't understand what's going on here. Same for `buildNodeSlice()`. I get what you are trying to do, but not the implementation.

Comment (spilchen) [reply pkg/atconf/file_writer.go:310]: We are picking the next available node number based on nodes[]. We update this slice as we go, which may involve extending it, so we always return a new copy of nodes[].

Any suggestions on how I can make it easier to understand?

Comment (roypaulin) [reply pkg/atconf/file_writer.go:310]: I think some "why" comments in `buildNodeSlice()` and `getNextNodeNumber()` would help. Especially where you extend the slice. And maybe adding, in `buildNodeSlice() description`, what we are going to do with the slice.

**Review by roypaulin — APPROVED** [review]
Looks good!

**Pull request merged**